srtcore/core.cpp

```diff
@@ -429,22 +440,22 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
         break;
 
     case SRTO_SNDSYN:
-        m_bSynSending = bool_int_value(optval, optlen);
+        m_bSynSending = cast_optval<bool>(optval, optlen);
         break;
 
     case SRTO_RCVSYN:
-        m_bSynRecving = bool_int_value(optval, optlen);
+        m_bSynRecving = cast_optval<bool>(optval, optlen);
         break;
 
     case SRTO_FC:
         if (m_bConnecting || m_bConnected)
             throw CUDTException(MJ_NOTSUP, MN_ISCONNECTED, 0);
 
-        if (*(int *)optval < 1)
+        if (cast_optval<int>(optval, optlen) < 1)
```

Comment:
Proposal: add here:
```
{
    int val = cast_optval(optval, optlen);

    if (val < 1)
...

    if (val > 32)
          m_iFlightFlagSize = val;
...
}
```

Reply:
I'm afraid SonarCloud also complains about such hanging code blocks, motivating to put the code into a separate function instead.

Reply:
An intermediate approach would be to declare variables of types used by options, initialize them with 0, then use the appropriate one inside the option handler. This will be no reason for the SC to complain, at least for that reason.

Reply:
I get your point of having variables of each type. But somehow I don't feel it is good here. 🤔